neurox/interpretation/linear_probe.py

```diff
@@ -89,6 +94,9 @@ def _train_probe(
     is trained with Cross Entropy loss for classification tasks and a linear
     regression model is trained with MSE loss for regression tasks. The
     optimizer used is Adam with default ``torch.optim`` hyperparameters.
+    On GPU, the probe is trained with mixed precision (amp).
```

Comment:
As i understand from your comment in the PR and here, GPU training requires float32, and CPU training as well. If the model is float32, and the batch is float32, why do we need mixed-precision training here?

Reply:
This is important, as the `scaler` will be used regardless if someone wants float16 or not, which changes the default behavior. I assume scaler with float32 won't change anything, but we need to be confident about this. Your insight and expertise is appreciated here!

Reply:
I added mixed precision training because I assumed it would slightly accelerate training. Thinking about it again, it is not strictly needed, and indeed the default behavior might change. I removed it again. That means that now, probe training always uses full precision by converting individual batches to float32 if necessary.